src/main/java/edu/kis/powp/jobs2d/drivers/CanvasManager.java

```diff
@@ -20,6 +23,17 @@ public synchronized void setCurrentCanvas(Canvas canvas) {
         changePublisher.notifyObservers();
     }
 
+    /**
+     * @param command commands
+     * @return value false or true if command can fit into current Canvas
+     */
+    public void checkCanvas(DriverCommand command)
+    {
+        CanvasVisitor canvasVisitor = new CanvasVisitor(this.currentCanvas);
+        command.accept(canvasVisitor);
```

Comment:
The result should bv checked after visitation and the appropriate action (e.g., warning in logs) should take place.